File: .cursor/skills/agents/references/examples/fastapi-app/src/routes/health.py
"""Health check endpoints."""

from fastapi import APIRouter

from src.config import settings

router = APIRouter()


@router.get("/health")
async def health_check() -> dict[str, str]:
    """Basic health check endpoint."""
    return {"status": "healthy", "app": settings.app_name}


@router.get("/ready")
async def readiness_check() -> dict[str, str]:
    """Readiness check endpoint."""
    # Could add database connectivity check here
    return {"status": "ready"}

File: .cursor/skills/agents/references/examples/express-api-ts/src/routes/health.ts
import { Router, type Request, type Response } from 'express';

const router = Router();

router.get('/', (_req: Request, res: Response) => {
  res.json({
    status: 'healthy',
    timestamp: new Date().toISOString(),
    uptime: process.uptime(),
  });
});

router.get('/ready', (_req: Request, res: Response) => {
  // Add database connectivity check here
  res.json({ ready: true });
});

export { router as healthRoutes };
